index.js

```diff
@@ -71,7 +87,11 @@ Collector.prototype.flush = function() {
 
     // only send N points at a time to avoid making requests too large
     // TODO what if we are backed up?
-    var points = self._points.splice(0, 50);
+
+    var spliceIndex = self.computePointCountToSend(self._points.map(function (point) {
+      return JSON.stringify(point).length;
```

Comment:
We don't send the data JSON formatted. This change might make more sense in the [influx-udp] (https://github.com/heap/node-influx-udp/tree/heap) package were we can compute the data size based on the message protocol used and by definition the change will only affect UDP endpoints.